source/extensions/common/wasm/stats_handler.h

```diff
@@ -89,17 +90,21 @@ CreateStatsHandler& getCreateStatsHandler();
 
 class LifecycleStatsHandler {
 public:
-  LifecycleStatsHandler(const Stats::ScopeSharedPtr& scope, std::string runtime)
+  LifecycleStatsHandler(const Stats::ScopeSharedPtr& scope, std::string runtime,
+                        std::string plugin_name)
       : lifecycle_stats_(LifecycleStats{
             LIFECYCLE_STATS(POOL_COUNTER_PREFIX(*scope, absl::StrCat("wasm.", runtime, ".")),
-                            POOL_GAUGE_PREFIX(*scope, absl::StrCat("wasm.", runtime, ".")))}){};
+                            POOL_GAUGE_PREFIX(*scope, absl::StrCat("wasm.", runtime, ".")),
+                            POOL_GAUGE_PREFIX(*scope, absl::StrCat("wasm.", runtime, ".plugin.",
+                                                                   plugin_name, ".")))}){};
   ~LifecycleStatsHandler() = default;
 
   void onEvent(WasmEvent event);
   static int64_t getActiveVmCount();
 
 protected:
   LifecycleStats lifecycle_stats_;
+  bool is_crashed_ = false;
```

Comment:
I believe `LifecycleStatsHandler`is the global object and `is_crashed_` shouldn't be bounded here, but instead vm-scoped object.

Reply:
Each instance of Envoy::Extensions::Common::Wasm::Wasm() has a separate lifecycleStatsHandler